e2e_playwright/st_dataframe_interactions_test.py

```diff
@@ -543,6 +545,60 @@ def test_row_hover_highlight(themed_app: Page, assert_snapshot: ImageCompareFunc
     assert_snapshot(df, name="st_dataframe-row_hover_highlight")
 
 
+def test_sorting_column_via_ui(app: Page, assert_snapshot: ImageCompareFunction):
+    """Test that a column can be sorted via the UI by clicking on the column
+    header and via the column menu."""
+    df = app.get_by_test_id("stDataFrame").nth(0)
+    expect_canvas_to_be_visible(df)
+
+    assert_snapshot(df, name="st_dataframe-no_sorting")
+
+    # Click on the column header to sort in ascending order:
+    click_on_cell(df, 0, 2, column_width="small")
+    unfocus_dataframe(app)
+    assert_snapshot(df, name="st_dataframe-sorted_ascending")
+
+    # Click on the column header again to sort in descending order:
+    click_on_cell(df, 0, 2, column_width="small")
+    unfocus_dataframe(app)
+    assert_snapshot(df, name="st_dataframe-sorted_descending")
+
+    # Click on the column header again to remove sorting:
+    click_on_cell(df, 0, 2, column_width="small")
+    unfocus_dataframe(app)
+    assert_snapshot(df, name="st_dataframe-no_sorting")
+
+    # Open the column menu and sort in ascending order:
```

Comment:
nit: after the steps above, why doing it again? If this is intended, probably good to leave a comment and perhaps put it in a loop? 🤔

Reply:
The steps above are sorting by click on header, and these steps are doing sorting via menu click using the same snapshots generated above (since the same state is expected).